gems/aws-sdk-s3/lib/aws-sdk-s3/presigner.rb

```diff
@@ -195,6 +195,15 @@ def sign_but_dont_send(
         req.handlers.remove(Aws::S3::Plugins::S3Signer::V4Handler)
         req.handlers.remove(Seahorse::Client::Plugins::ContentLength::Handler)
 
+        # Used for excluding presigned_urls from API request count.
+        #
+        # Store context information as early as possible, to allow
+        # handlers to perform decisions based on this flag if need.
+        req.handle(step: :initialize, priority: 98) do |context|
+          context[:presigned_url] = true
+          @handler.call(context)
+        end
+
```

Comment:
Could you move this to its own method, similar to `use_bucket_as_hostname` and call it from `_presigned_request`?